### ISSUE TITLE
FN implicit if_same_then_else

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rust-clippy#3030]
````rust
fn main() {
	println!("{:?}", func(4));
}

fn func(n: i32) -> bool {
	if n < 2 {
		return true;
	} else if  n > 2 {
		return false;
	}
	return false; // return outside of conditions
}
````
this code is basically identical to
````rust
fn main() {
	println!("{:?}", func(4));
}

fn func(n: i32) -> bool {
	if n < 2 {
		return true;
	} else if  n > 2 {
		return false;
	} else { // return moved into else {}
		return false;
	}
}
````
but only the latter triggers the ````if_same_then_else```` warning.
Should the first example trigger a warning as well?

**Comment by felix91gr** [comment]
I haven't been able to trigger the lint again with any of those examples. 

Both do trigger the `needless_return` lint:

* Example 1: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=723da7d949606cdc2ea45b5df999598e

* Example 2: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=d8dd4b882a69f4dce8bcb33fb7861e5a

If we do as the `needless_return` lint says, we go back to the behavior you observed, @matthiaskrgr.

* The first example (after removing the last `return`) doesn't trigger the lint: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=810d75928f41a8bfeff1fcf97f505c1f

```rust
fn main() {
	println!("{:?}", func(4));
}

fn func(n: i32) -> bool {
	if n < 2 {
		return true;
	} else if  n > 2 {
		return false;
	}
	return false; // return outside of conditions
}
```

* Whereas the second (after removing all three `return` statements) does trigger it: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=b47bc99288facc7bb64f22d6a77faa2f

```rust
fn main() {
	println!("{:?}", func(4));
}

fn func(n: i32) -> bool {
	if n < 2 {
		true
	} else if  n > 2 {
		false
	} else {
	    false // return outside of conditions
    }
}
```

So I'd say the bug still stands as reported.
